### PR TITLE
Feature/correct settle payment

### PR DESCRIPTION
**Pull request by falklinus**
# Jira Issue: [CLX-261]

## What?
- "Correct" settled automatic payment by creating manual payment order with negative amount and deductible 
- Show information about potential correction
## Why?
- Wanted feature by finance

## Optional screenshots
image
image
image

## Optional checklist
- [ ] Updated `src/changelog.ts`
- [ ] Codescouted
- [ ] Unit tests written
- [x] Tested locally

[CLX-261]: https://hedvig.atlassian.net/browse/CLX-261?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/portals/hope/pages/tools/payment-orders/hooks/use-has-been-corrected.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        amount < 0,
+    )?.id ?? null
+
+  const hasBeenCorrected = loading ? null : !!correctingOrder
+
+  return {
+    hasBeenCorrected,
+    correctingOrder,
+  }
+}
```
Comment (falklinus) [inline src/portals/hope/pages/tools/payment-orders/hooks/use-has-been-corrected.ts:41]: Remove and use information directly from paymentOrder, once it is available

File: src/portals/hope/pages/tools/payment-orders/hooks/use-corrects-order.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { PaymentOrderInformationFragment } from 'types/generated/graphql'
+import { PaymentMethod } from 'portals/hope/features/config/constants'
+
+export const useCorrectsOrder = (order?: PaymentOrderInformationFragment) => {
```
Comment (falklinus) [inline src/portals/hope/pages/tools/payment-orders/hooks/use-corrects-order.ts:4]: Remove and use information directly from paymentOrder, once it is available

File: src/portals/hope/pages/tools/payment-orders/components/OrderStateActions.tsx [hunk 3]
```diff
@@ -1,52 +1,113 @@
...
+              amount: -order.deductible.amount,
+              currency: order.deductible.currency,
+            },
+            isExGratia: order.isExGratia,
+            sanctionListSkipped: order.sanctionListSkipped,
+            note: `Correction for order ${order.id}`,
+            recipientType: order.recipientType,
+            recipientName: order.recipientName,
+            reference: order.reference,
+          })
```
Comment (falklinus) [inline src/portals/hope/pages/tools/payment-orders/components/OrderStateActions.tsx:62]: Change to be same as original order (keep negative amount and deductible)

File: src/portals/hope/pages/tools/payment-orders/components/OrderStateActions.tsx [hunk 4]
```diff
@@ -1,52 +1,113 @@
...
-    usePaymentOrder()
+  const {
+    createPaymentOrder,
+    settlePaymentOrder,
+    dismissPaymentOrder,
+    approvePaymentOrder,
+  } = usePaymentOrder()
+  const [makeCorrection, setMakeCorrection] = useState(false)
+
+  const { hasBeenCorrected } = useHasBeenCorrected(order)
```
Comment (falklinus) [inline src/portals/hope/pages/tools/payment-orders/components/OrderStateActions.tsx:27]: can come directly from order once available

File: src/portals/hope/pages/tools/payment-orders/components/OrderInformation.tsx [hunk 5]
```diff
@@ -17,6 +20,8 @@ export const OrderInformation = ({
   order: PaymentOrderInformationFragment
   standalone: boolean
 }) => {
+  const { hasBeenCorrected, correctingOrder } = useHasBeenCorrected(order)
+  const { correctsOtherPayment, correctedOrderId } = useCorrectsOrder(order)
```
Comment (falklinus) [inline src/portals/hope/pages/tools/payment-orders/components/OrderInformation.tsx:24]: can come directly from order once available

**Review by vonElfvin — APPROVED** [review]
💥

**Pull request merged**